pallets/oracle/src/dia.rs

```diff
@@ -30,8 +48,16 @@ impl Convert<OracleKey, Option<(Vec<u8>, Vec<u8>)>> for DiaOracleKeyConvertor {
 					_ => unimplemented!(),
 				},
 				CurrencyId::Native => unimplemented!(),
-				CurrencyId::StellarNative => unimplemented!(),
-				CurrencyId::Stellar(primitives::Asset::AlphaNum4 { .. }) => unimplemented!(),
+				CurrencyId::StellarNative => Some((
+					STELLAR_DIA_BLOCKCHAIN.as_bytes().to_vec(),
+					STELLAR_DIA_SYMBOL.as_bytes().to_vec(),
+				)),
+				CurrencyId::Stellar(primitives::Asset::AlphaNum4 { code, .. }) => {
+					let fiat_quote =
+						construct_fiat_usd_symbol_for_currency(code.to_ascii_uppercase());
```

Comment:
i guess the `to_ascii_uppercase()` call is not necessary, since the function itself will do it too.